pkg/new-kernel/Dockerfile

```diff
@@ -93,6 +96,7 @@ ENV KERNEL_DEFCONFIG=defconfig
 # build for all arches
 # hadolint ignore=DL3006
 FROM kernel-target-${TARGETARCH} AS kernel-build
+ARG BUILDARCH
```

Comment:
@famleebob, you forgot this ARG from the first version of the PR, we don't need it anymore (correctly pointed out by @christoph-zededa)